data-prepper-plugins/ml-inference-processor/src/main/java/org/opensearch/dataprepper/plugins/ml/processor/MLProcessor.java

```diff
@@ -0,0 +1,95 @@
+import org.opensearch.dataprepper.model.annotations.Experimental;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.processor.AbstractProcessor;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.ml.processor.common.MLBatchJobCreator;
+import org.opensearch.dataprepper.plugins.ml.processor.common.MLBatchJobCreatorFactory;
+import org.opensearch.dataprepper.plugins.ml.processor.configuration.ServiceName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.opensearch.dataprepper.logging.DataPrepperMarkers.NOISY;
+
+@Experimental
+@DataPrepperPlugin(name = "ml_inference", pluginType = Processor.class, pluginConfigurationType = MLProcessorConfig.class)
+public class MLProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+    public static final Logger LOG = LoggerFactory.getLogger(MLProcessor.class);
+    public static final String NUMBER_OF_ML_PROCESSOR_SUCCESS = "mlProcessorSuccessfullyCreated";
+    public static final String NUMBER_OF_ML_PROCESSOR_FAILED = "mlProcessorFailedToCreated";
+
+    private final String whenCondition;
+    private MLBatchJobCreator mlBatchJobCreator;
+    private final Counter numberOfMLProcessorSuccessCounter;
+    private final Counter numberOfMLProcessorFailedCounter;
+    private final ExpressionEvaluator expressionEvaluator;
+
+    @DataPrepperPluginConstructor
+    public MLProcessor(final MLProcessorConfig mlProcessorConfig, final PluginMetrics pluginMetrics, final AwsCredentialsSupplier awsCredentialsSupplier, final ExpressionEvaluator expressionEvaluator) {
+        super(pluginMetrics);
+        this.whenCondition = mlProcessorConfig.getWhenCondition();
+        ServiceName serviceName = mlProcessorConfig.getServiceName();
+        this.numberOfMLProcessorSuccessCounter = pluginMetrics.counter(
+                NUMBER_OF_ML_PROCESSOR_SUCCESS);
+        this.numberOfMLProcessorFailedCounter = pluginMetrics.counter(
+                NUMBER_OF_ML_PROCESSOR_FAILED);
+        this.expressionEvaluator = expressionEvaluator;
+
+        // Use factory to get the appropriate job creator
+        mlBatchJobCreator = MLBatchJobCreatorFactory.getJobCreator(serviceName, mlProcessorConfig, awsCredentialsSupplier, pluginMetrics);
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
+        // reads from input - S3 input
+        if (records.size() == 0)
+            return records;
+
+        List<Record<Event>> recordsToMlCommons = records.stream()
+                .filter(record -> whenCondition == null || expressionEvaluator.evaluateConditional(whenCondition, record.getData()))
+                .collect(Collectors.toList());
```

Comment:
I would recommend that you catch exceptions for this `evaluateConditional` call. There can be things that cause this to throw, so better to not have the processor throw exception and drop the full batch of records (if expression throws, then could be a good idea to log exception and filter it out from this list

Reply:
good catch